.github/workflows/push-actions.yml

```diff
@@ -1,12 +1,38 @@
 name: Tracer Testing
 on: [push]
 jobs:
+  list-instrumentations:
+    name: List supported instrumentations
+    runs-on: ubuntu-latest
+    outputs:
+      integration-tests: ${{ steps.discover_itests.outputs.integration_tests }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          architecture: x64
+      - name: Install jq
+        run:
+          sudo apt-get update && sudo apt-get install -y jq
+      - name: Install Python dependencies
+        run: |
+          pip install -r requirements.txt
+      - name: Generate list of integration tests
+        id: discover_itests
+        run: |
+          echo "integration_tests=$(nox --list | grep 'integration_tests_' | awk '{ print $2 }' | sed -r 's|^integration_tests_([^-]*)(-[[:digit:]].[[:digit:]]+)?\(.*\)$|\1|' | uniq | sort | jq -Rnc '[inputs]')" | tee /dev/stderr >> $GITHUB_OUTPUT
```

Comment:
can you re-write these lines in a more readable way?

Reply:
a61902668722bca37da604878c3c497c8eae5b53 is a masterpiece, if I say so myself :D